cmd/thv-registry-api/app/serve.go

```diff
@@ -90,85 +88,97 @@ func getKubernetesConfig() (*rest.Config, error) {
 	return kubeConfig.ClientConfig()
 }
 
-// buildProviderConfig creates provider configuration based on command-line flags
-func buildProviderConfig() (*service.RegistryProviderConfig, error) {
-	configMapName := viper.GetString("from-configmap")
-	filePath := viper.GetString("from-file")
-	registryName := viper.GetString("registry-name")
+// createFileProviderForSyncedData creates a file provider that reads from the synced storage location
+// All source types (git, configmap, api, file) are synced to ./data/registry.json by the sync manager
+func createFileProviderForSyncedData(registryName string) (*service.RegistryProviderConfig, error) {
+	return &service.RegistryProviderConfig{
+		FilePath:     "./data/registry.json",
+		RegistryName: registryName,
+	}, nil
+}
 
-	// Validate mutual exclusivity
-	if configMapName != "" && filePath != "" {
-		return nil, fmt.Errorf("--from-configmap and --from-file flags are mutually exclusive")
+func runServe(_ *cobra.Command, _ []string) error {
+	ctx := context.Background()
+	address := viper.GetString("address")
+
+	// Initialize controller-runtime logger to suppress warnings
+	log.SetLogger(zap.New(zap.UseDevMode(false)))
+
+	logger.Infof("Starting registry API server on %s", address)
+
+	// Load and validate configuration (now required)
+	configPath := viper.GetString("config")
+	// TODO: Validate the path to avoid path traversal issues
+	cfg, err := config.NewConfigLoader().LoadConfig(configPath)
+	if err != nil {
+		return fmt.Errorf("failed to load configuration: %w", err)
 	}
+	if err := cfg.Validate(); err != nil {
+		return fmt.Errorf("invalid configuration: %w", err)
+	}
+	logger.Infof("Loaded configuration from %s (registry: %s, source: %s)",
+		configPath, cfg.GetRegistryName(), cfg.Source.Type)
 
-	// Require one of the flags
-	if configMapName == "" && filePath == "" {
-		return nil, fmt.Errorf("either --from-configmap or --from-file flag is required")
+	// Get Kubernetes config (needed for both sync manager and deployment provider)
+	k8sRestConfig, err := getKubernetesConfig()
+	if err != nil {
+		logger.Warnf("Failed to create kubernetes config: %v", err)
+		logger.Warn("ConfigMap source and deployment provider will not be available")
 	}
 
-	// Require registry name
-	if registryName == "" {
-		return nil, fmt.Errorf("--registry-name flag is required")
+	// Initialize sync manager for automatic registry synchronization
+	logger.Info("Initializing sync manager for automatic registry synchronization")
+
+	// Create Kubernetes scheme and register core types
+	scheme := runtime.NewScheme()
+	if err := clientgoscheme.AddToScheme(scheme); err != nil {
+		return fmt.Errorf("failed to add Kubernetes core types to scheme: %w", err)
 	}
 
-	if configMapName != "" {
-		config, err := getKubernetesConfig()
+	// Create Kubernetes client (may be nil if k8s not available)
+	var k8sClient client.Client
```

Comment:
1We are creating a lot of clients, schemes and variables directly in serve.go. I think it would be nicer to define a structure (maybe called `RegistryApp` or such) that would be able to build the server from components (sourceHandlerFactory, storageManager, statusPersistence). The default would be to build everything from config but e.g. tests could pass their own mocks. The default could build the app from config, k8s client, scheme etc.